### PR TITLE
Add precompilation

### LINKED ISSUE #402: Latency
Could be good to run https://discourse.julialang.org/t/new-tools-for-reducing-compiler-latency/52882, although I would imagine that our latency cost is because of our deps, not because of us.

### PR DESCRIPTION
**Pull request by antoine-levitt**
Closes #402.

### CODE REVIEW

File: src/DFTK.jl [hunk 1]
```diff
@@ -232,4 +232,22 @@ function __init__()
...
+# Precompilation block with a basic workflow
+using SnoopPrecompile
+if VERSION ≥ v"1.9alpha" && isnothing(get(ENV, "DFTK_NO_PRECOMPILATION", nothing))
+    @precompile_all_calls begin
+        # simple silicon ground state example
+        a = 10.26
+        lattice = a / 2 * [[0 1 1.];
+                           [1 0 1.];
+                           [1 1 0.]]
+        Si = ElementPsp(:Si, psp=load_psp("hgh/lda/Si-q4"))
```
Comment (mfherbst) [inline src/DFTK.jl:244]: Does it need to be as complicated as silicon? Why don just a helium atom in a box or so?

File: src/DFTK.jl [hunk 2]
```diff
@@ -232,4 +232,22 @@ function __init__()
...
+        atoms     = [Si, Si]
+        positions = [ones(3)/8, -ones(3)/8]
+
+        model = model_LDA(lattice, atoms, positions)
+        basis = PlaneWaveBasis(model; Ecut=15, kgrid=[4, 4, 4])
```
[the remark this answers is not in the document]
Comment (antoine-levitt) [reply src/DFTK.jl:249]: This is 1s, so runtime is very reasonable. I just went with the simplest thing, I'm not sure how transferable it is to other workflows.

File: src/DFTK.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/DFTK.jl:244]: Why not? This loads psps, symmetries, etc.

**Comment by antoine-levitt** [comment]
metallic_systems.jl runs in 16s (that's including the Plots call), so still pretty good even if we don't precompile its features specifically.

File: src/DFTK.jl [hunk 3]
```diff
@@ -232,4 +232,22 @@ function __init__()
...
+        scfres = self_consistent_field(basis, tol=1e-2)
```
[the remark this answers is not in the document]
Comment (mfherbst) [reply src/DFTK.jl:250]: I'm just thinking to put the least effort into the precompilation stage.

File: src/DFTK.jl (hunk 1 above)
Comment (mfherbst) [reply src/DFTK.jl:244]: ok.

File: src/DFTK.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (mfherbst) [reply src/DFTK.jl:249]: yeah but enabling temperature or multiple spin channels cannot hurt and is not even a line of code.

File: src/DFTK.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (mfherbst) [reply src/DFTK.jl]: Ah tol=10 has the disadvantage that the mixing might not run. That's why I suggested maxiter=3, maxiter=2 would probably also cut it.

**Pull request merged**